src/core/namespace.c

```diff
@@ -599,11 +599,26 @@ static int clone_device_node(const char *d, const char *temporary_mount) {
         dn = strjoina(temporary_mount, d);
 
         mac_selinux_create_file_prepare(d, st.st_mode);
-        r = mknod(dn, st.st_mode, st.st_rdev);
+        if (do_mknod)
+                r = mknod(dn, st.st_mode, st.st_rdev);
+        else
+                r = mknod(dn, S_IFREG, st.st_rdev);
         mac_selinux_create_file_clear();
-        if (r < 0)
+        if (r < 0 && (do_mknod || errno != EEXIST))
                 return log_debug_errno(errno, "mknod failed for %s: %m", d);
 
+        /* Fallback to bind-mounting:
+         * The assumption here is that all used device nodes carry standard
+         * properties. Specifically, the devices nodes we bind-mount should
+         * either be owned by root:root or root:tty (e.g. /dev/tty, /dev/ptmx)
+         * and should not carry ACLs.
+         */
+        if (!do_mknod) {
+                r = mount(d, dn, NULL, MS_BIND, NULL);
+                if (r < 0)
+                        return log_debug_errno(errno, "%s failed for %s: %m", do_mknod ? "mknod" : "mount", d);
```

Comment:
the ternary expression appears unnecessary, no? we check do_mknod first anyway

Reply:
ugh, duh. sorry
addressed